test/unit/common/viz/FigureExtractor.spec.js

```diff
@@ -0,0 +1,52 @@
+describe('FigureExtractor', function() {
+    const testFixture = require('../../../globals');
+    const assert = require('assert');
+    const fs = require('fs');
+    const path = require('path');
+    const FigureExtractor = testFixture.requirejs('deepforge/viz/FigureExtractor').CoreFigureExtractor;
+    const gmeConfig = testFixture.getGmeConfig();
+    const logger = testFixture.logger.fork('FigureExtractor');
+    const GRAPH_NODE_PATH = '/K/n/d/2';
+    const REFERENCE_JSON = path.resolve(__dirname, 'FigureReference.json');
+    let project,
+        gmeAuth,
+        storage,
+        commitHash,
+        rootNode,
+        core;
+
+    before(async function () {
+        const projectName = 'testProject';
+        gmeAuth = await testFixture.clearDBAndGetGMEAuth(gmeConfig, projectName);
+        storage = testFixture.getMemoryStorage(logger, gmeConfig, gmeAuth);
+        await storage.openDatabase();
+        const importParam = {
+            projectSeed: testFixture.path.join(testFixture.DF_SEED_DIR, 'devProject', 'devProject.webgmex'),
+            projectName: projectName,
+            branchName: 'master',
+            logger: logger,
+            gmeConfig: gmeConfig
+        };
+
+        const importResult = await testFixture.importProject(storage, importParam);
+        project = importResult.project;
+        commitHash = importResult.commitHash;
+        core = importResult.core;
+        rootNode = importResult.rootNode;
+        await project.createBranch('test', commitHash);
+    });
+
+    it('should convert graphNode to JSON', async () => {
```

Comment:
I have added the test. Essentailly, the fixture below tested it, but didnot take into account the `cache`.

Reply:
It would be good to check that they reference the same thing actually. This should implicitly be testing the cache.
```javascript
assert(graphNodeJSON === child.parent);
```

Reply:
I almost missed this line.
https://github.com/deepforge-dev/deepforge/blob/c00efca1504464c451249b63df67b59f74387199/src/common/viz/FigureExtractor.js#L300
 Thank You.